File: leetcode/121.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        
        diff = 0
        left = prices[0]

        for i in range(1, len(prices)):
            cur = prices[i]
            if cur < left:
                left = cur
            diff = max(diff, cur - left)

        return diff
                





        

File: leetcode/121.cpp
class Solution {
  public:
  int maxProfit(vector<int>& prices)
  {
    int profit, price, n, i;
    profit = 0;
    price = prices[0];

    n = prices.size();

    for (i = 1; i < n; ++i) {
      if (prices[i] > price) {
        profit = max(profit, prices[i] - price);
      } else {
        price = min(price, prices[i]);
      }
    }
    return profit;
  }
};
